napari/layers/labels/_labels_constants.py

```diff
@@ -35,4 +35,17 @@ class Mode(StringEnum):
     ERASE = auto()
 
 
+class ColorMode(StringEnum):
+    """
+    ColorMode: Color setting mode.
+
+    DIRECT (default mode) allows each color to be set arbitrarily
+
+    DICT allows color to be set via a color dictionary
```

Comment:
```suggestion
    RANDOM (default mode) allows color to be set via a hash function with a seed.

    DIRECT allows color of each label to be set directly by a color dictionary.
```

I think I'd probably swap what we call DICT and DIRECT and use RANDOM, but curious what @kevinyamauchi thinks. Also what would be name the third mode where we allow settings based on a property?

Reply:
I agree we should call the "allows color to be set via a color dictionary" mode `DIRECT`.

I am not sure what the best name would be for the "(default mode) allows color to be set via a hash function with a seed" mode. I think `RANDOM` is okay, but I don't think I would know how to interpret it without the docstring. Maybe `AUTO` and then include a slight variant of the description @sofroniewn suggested: "automatically assigns label colors via a hash function with a seed"

For setting with a property, I imagine we might want to allow the user to have an automated mapping between property value and color and also to set the mapping directly (see #1348). If that is the case, maybe `DIRECT_PROPERTY` and `AUTO_PROPERTY`?

Reply:
One more question: sorry for not catching this earlier, but should we name the Enum something different. We use `ColorMode` in the same way in the `Points`, `Vectors` and soon to be `Shapes` layers. Since this has a different usage, perhaps it would be best to give it a different name.

Maybe something like `LabelColorMode`? Too long? Sorry, I'm not good at naming...

edit: or `LabelsColorMode`?

Reply:
I was hoping that we could unify the practices and later using a single enum. But for now I can rename it